src/main/java/com/amazon/opendistro/elasticsearch/performanceanalyzer/rca/configs/DeciderConfig.java

```diff
@@ -0,0 +1,120 @@
+    // Defaults based on prioritising Stability over performance.
+    private static final List<String> DEFAULT_WORKLOAD_PRIORITY = Arrays.asList("ingest", "search");
+    private static final List<String> DEFAULT_CACHE_PRIORITY = Arrays.asList("read-heavy", "write-heavy");
+
+    private Double fieldDataCacheUpperBound;
+    private Double shardRequestCacheUpperBound;
+    private List<String> cachePriorityOrder;
+    private List<String> workloadPriorityOrder;
+
+    public DeciderConfig(final RcaConf rcaConf) {
+        fieldDataCacheUpperBound = rcaConf.readDeciderConfig(CACHE_BOUNDS_CONFIG_NAME,
+                FIELD_DATA_CACHE_UPPER_BOUND_NAME, Double.class);
+        shardRequestCacheUpperBound = rcaConf.readDeciderConfig(CACHE_BOUNDS_CONFIG_NAME,
+                SHARD_REQUEST_CACHE_UPPER_BOUND_NAME, Double.class);
+        cachePriorityOrder = rcaConf.readDeciderConfig(CACHE_CONFIG_NAME,
+                PRIORITY_ORDER_CONFIG_NAME, List.class);
+        workloadPriorityOrder = rcaConf.readDeciderConfig(WORKLOAD_CONFIG_NAME,
+                PRIORITY_ORDER_CONFIG_NAME, List.class);
+        if (fieldDataCacheUpperBound == null) {
+            fieldDataCacheUpperBound = DEFAULT_FIELD_DATA_CACHE_UPPER_BOUND;
+        }
+        if (shardRequestCacheUpperBound == null) {
+            shardRequestCacheUpperBound = DEFAULT_SHARD_REQUEST_CACHE_UPPER_BOUND;
+        }
+        if (cachePriorityOrder == null) {
+            cachePriorityOrder = DEFAULT_CACHE_PRIORITY;
+        }
+        if (workloadPriorityOrder == null) {
+            workloadPriorityOrder = DEFAULT_WORKLOAD_PRIORITY;
+        }
+    }
+
+    public Double getFieldDataCacheUpperBound() {
+        return fieldDataCacheUpperBound;
+    }
+
+    public Double getShardRequestCacheUpperBound() {
+        return shardRequestCacheUpperBound;
+    }
+
+    public List<String> getCachePriorityOrder() {
+        return cachePriorityOrder;
+    }
+
+    public List<String> getWorkloadPriorityOrder() {
+        return workloadPriorityOrder;
+    }
+
+    public static List<String> getDefaultWorkloadPriority() {
+        return DEFAULT_WORKLOAD_PRIORITY;
+    }
+
+    public static List<String> getDefaultCachePriority() {
+        return DEFAULT_CACHE_PRIORITY;
+    }
+
+    public static Double getDefaultShardRequestCacheUpperBound() {
+        return DEFAULT_SHARD_REQUEST_CACHE_UPPER_BOUND;
+    }
+
+    public static Double getDefaultFieldDataCacheUpperBound() {
+        return DEFAULT_FIELD_DATA_CACHE_UPPER_BOUND;
+    }
+
+    public static String getCacheBoundsConfigName() {
+        return CACHE_BOUNDS_CONFIG_NAME;
+    }
+
+    public static String getCacheConfigName() {
+        return CACHE_CONFIG_NAME;
+    }
+
+    public static String getWorkloadConfigName() {
+        return WORKLOAD_CONFIG_NAME;
+    }
+
+    public static String getPriorityOrderConfigName() {
+        return PRIORITY_ORDER_CONFIG_NAME;
+    }
+
+    public static String getFieldDataCacheUpperBoundName() {
```

Comment:
I have created getters for all the private fields. We don't need them now but maybe if want to re-use the variables in some other module, it would be nice to keep it here.